py/_io/terminalwriter.py

```diff
@@ -144,12 +144,30 @@ def __init__(self, file=None, stringio=False, encoding=None):
     def fullwidth(self):
         if hasattr(self, '_terminal_width'):
             return self._terminal_width
-        return get_terminal_width()
+        try:
+            width = self._cached_terminal_width
+        except AttributeError:
+            try:
+                signal.signal(signal.SIGWINCH, self._handle_sigwinch)
+            except AttributeError:  # SIGWINCH not supported.
+                width = self._cached_terminal_width = False
```

Comment:
So a registry for terminalwriter handlers might make sense, which then gets invalided on SIGWINCH (through a single registered callback)?